packages/actor-extract-links-all/lib/ActorExtractLinksAll.ts

```diff
@@ -19,7 +27,7 @@ export class ActorExtractLinksAll extends ActorExtractLinks {
     return {
       links: await ActorExtractLinks.collectStream(action.metadata, (quad, links) => {
         for (const link of getNamedNodes(getTerms(quad))) {
-          links.push({ url: link.value });
```

Comment:
The current approach with `annotateLinkWithTheReachabilityCriteria` and `reachabilityLabel` seems a bit complicated.

Could we just do something like `links.push({ url: link.value, metadata: { actorExtractLinks: this.name } })`?

Reply:
I was not using the name because some actors such has predicate have multiple common variation (common, ldp, solidStorage) and I felt this was it was capture better. Also I wanted to make the annotation optional hence why it is inside a method to not add to much if statement and since it will be same everywhere I placed it inside the abstract class.

Reply:
> some actors such has predicate have multiple common variation (common, ldp, solidStorage)

Perhaps those can be passed as additional metadata fields?

The main concern I have with this approach is that all actors have a new (optional) param now, and there is semantic redundancy with the name of the actor.

Reply:
I could create a reachability criteria object documenting the specificity of the criteria (with custom equal function for when you care about the "major" reachability type or the full reachability). But I think the optional parameters is another question, the reason it is there is to not modify the previous behavior where the metadata was empty. I defined this new parameter in the bus/abstract class to not disturbed too much the current and future implementation. I could make it so it is always in the links.